docs/source/daml2ts/index.rst

```diff
@@ -0,0 +1,290 @@
++-------------------+------------------+----------------------------------+
+| ``Decimal``       | ``Decimal``      | ``string``                       |
++-------------------+------------------+----------------------------------+
+| ``Numeric``       | ``Numeric n``    | ``string``                       |
++-------------------+------------------+----------------------------------+
+| ``Text``          | ``Text``         | ``string``                       |
++-------------------+------------------+----------------------------------+
+| ``Time``          | ``Time``         | ``string``                       |
++-------------------+------------------+----------------------------------+
+| ``Party``         | ``Party``        | ``string``                       |
++-------------------+------------------+----------------------------------+
+| ``List<T>``       | ``[t]``          | ``T[]``                          |
++-------------------+------------------+----------------------------------+
+| ``Date``          | ``Date``         | ``string``                       |
++-------------------+------------------+----------------------------------+
+| ``ContractId<T>`` | ``ContractId t`` | ``string``                       |
++-------------------+------------------+----------------------------------+
+| ``Optional<T>``   | ``Optional t``   | ``null | (null extends T ?``     |
+|                   |                  | ``[] | [Exclude<T, null>] : T)`` |
++-------------------+------------------+----------------------------------+
+| ``TextMap<T>``    | ``TextMap t``    | ``{ [key: string]: T }``         |
++-------------------+------------------+----------------------------------+
+
+.. note::
+   The TypeScript types ``Time``, ``Decimal``, ``Numeric`` and ``Int`` all alias to ``string``. These choices relate to the avoidance of precision loss under serialization over the `json-api <../json-api/index.html>`_.
+
+.. note::
+   The Typescript definition of type ``Optional<T>`` in the above table might look complicated. It accounts for differences in the encoding of optional values when nested versus when they are not (i.e. "top-level"). For example, ``null`` and ``"foo"`` are two possible values of ``Optional<Text>`` whereas, ``[]`` and ``["foo"]`` are two possible values of type ``Optional<Optional<Text>>`` (``null`` is another possible value, ``[null]`` is **not**).
+
+DAML to TypeScript mappings
+---------------------------
+
+The mappings from DAML to TypeScript are best explained by example.
+
+Records
+~~~~~~~
+
+In DAML, we might model a person like this.
+
+.. code-block:: daml
+   :linenos:
+
+   data Person =
+     Person with
+       name: Text
+       party: Party
+       age: Int
+
+Given the above definition, the generated TypeScript code will be as follows.
+
+.. code-block:: typescript
+   :linenos:
+
+   type Person = {
+     name: string;
+     party: daml.Party;
+     age: daml.Int;
+   }
+
+Tuples
```

Comment:
We should move tuples up into the big table. This section is more about variants that about tuples.

Reply:
I'm not clear on how to do that. I thought I did a pretty good job on this (slightly challenging) section. Do you feel strongly about this?

Reply:
To me, it feels like we're in a section about data type definitions here. That's why tuples don't fit well. You can't define them yourself but they are defined in the stdlib. I guess that's what made this section challenging.

I suggest we add rows to the big table above:
```
+-----------+-------------------------+
| DAML      | TypeScript              |
+-----------+-------------------------+
| (a, b)    | { _1: a; _2: b }        |
| (a, b, c) | { _1: a; _2: b; _3: c } |
+-----------+-------------------------+
```
 And also add a remark that `n`-tuples for `n > 3` work analogously.

Reply:
Got it. Moving to the big table!